src/treeview_sandbox.ml

```diff
@@ -79,6 +79,88 @@ module Command = struct
     Extension_commands.register
       ~id:Extension_consts.Commands.open_sandbox_documentation handler
 
+  let _generate_documentation =
+    let handler (instance : Extension_instance.t) ~args =
+      let (_ : unit Promise.t) =
+        let open Promise.Syntax in
+        let sandbox = Extension_instance.sandbox instance in
+        let* odig = Odig.of_sandbox sandbox in
+        match odig with
+        | Error error ->
+          let (_ : 'a option Promise.t) =
+            Window.showErrorMessage ~message:error ()
+          in
+          Promise.resolve ()
+        | Ok odig -> (
+          let arg = List.hd_exn args in
+          let dep = Dependency.t_of_js arg in
+          let package_name = Sandbox.Package.name dep in
+          let options =
+            ProgressOptions.create ~location:(`ProgressLocation Notification)
+              ~title:
+                (Printf.sprintf "Generating documentation for %s" package_name)
+              ~cancellable:false ()
+          in
+          let task ~progress:_ ~token:_ = Odig.odoc_exec odig ~package_name in
+          let* result =
+            Vscode.Window.withProgress
+              (module Interop.Js.Result (Interop.Js.String) (Interop.Js.String))
+              ~options ~task
+          in
+          match result with
+          | Error _ ->
+            let (_ : 'a option Promise.t) =
+              Window.showErrorMessage
+                ~message:
+                  (Printf.sprintf "Error while generating documentation for %s"
+                     package_name)
```

Comment:
There should be a util fn in `import.ml`: `show_message`, which can be useful to replace this piece of code